minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterImpl.java

```diff
@@ -327,6 +334,13 @@ public MiniAccumuloClusterImpl(MiniAccumuloConfigImpl config) throws IOException
 
     this.config = config.initialize();
 
+    if (Boolean.TRUE.equals(Boolean
+        .valueOf(this.config.getSiteConfig().get(Property.TSERV_NATIVEMAP_ENABLED.name())))) {
+      if (!NativeMap.isLoaded())
+        throw new RuntimeException(
+            "MAC configured to use native maps, but unable to load the library.");
+    }
+
```

Comment:
I wouldn't expect this to work. The native maps are loaded by adding LD_LIBRARY_PATH environment to the tserver Process before it is launched. There's no reason to expect the native map to be loaded inside the MiniAccumuloClusterImpl... only for the processes it launches.

(Also, no need to do `if (Boolean.TRUE.equals(someBoolExpr))` when you can just do `if (someBoolExpr)`)

Reply:
`ConfigurableMacBase.createMiniAccumulo` sets `TSERV_NATIVEMAP_ENABLED` to true and then starts MAC. This checks that NativeMap will be able to load the library in this test thread before starting up the processes.

Reply:
Addressed logic comment in 3f7968f

Reply:
> This checks that NativeMap will be able to load the library in this test thread before starting up the processes.

There's no reason to believe that this test thread should be able to load the library. It doesn't have the native maps on its linker path. It's ability, or inability, to load native maps has no bearing on whether the tserver process that mini launches will be able to do so. They have different environments.

> `ConfigurableMacBase.createMiniAccumulo` sets `TSERV_NATIVEMAP_ENABLED` to true and then starts MAC.

Yes, and the way this works is that `MiniAccumuloClusterImpl` sets the `LD_LIBRARY_PATH` in the environment for the tserver's `Process`. Nothing does that for the test thread's environment. There's no reason to expect it to be able to load native maps.

Reply:
After fixing the bug where the config was being checked to see if the native maps were enabled (using `Property.name()` instead of `Property.getKey()`), I was able to confirm that this does, in fact, fail because the native maps can't be loaded in the test thread. `mvn clean verify -Dtest=blah -Dit.test=WALSunnyDayIT` is an easy one to use to reproduce this, since it extends `ConfigurableMacBase`, and doesn't disable the native maps.

So, this native map loading check in the test thread really needs to be removed for this to have any chance of verifyUp working properly.

Reply:
I will remove this morning. I was waiting on the results of an IT run after my changes in 3f7968f. In an IT run from the commit, only one test failed and it was unrelated to verifyUp.

Reply:
This inspired me to create #2425 , which will accomplish what you were trying to do inside Mini, but does it inside ConfigurableMacBase, among other improvements. I think there's still something that can be done here, though.

This could become something like:

```java
    if (Boolean.valueOf(config.getSiteConfig().get(Property.TSERV_NATIVEMAP_ENABLED.getKey()))
        && config.getNativeLibPaths().length == 0
        && !config.getSystemProperties().containsKey("accumulo.native.lib.path")) {
      throw new RuntimeException(
          "MAC configured to use native maps, but native library path was not provided.");
    }
```

Reply:
Regarding #2425, when I was looking at NativeMap, I was wondering why we first looked in `accumulo.native.lib.path` and then tried LD_LIBRARY_PATH as a fallback. It seems that this is another case where we are doing something non-standard. Why not just use LD_LIBRARY_PATH ?

Reply:
tl;dr there's good reasons behind it, but it's complicated.

Messing with `LD_LIBRARY_PATH` isn't good practice. Ideally, the native library is installed by a package manager into `/usr/lib{64}/whatever` and can be loaded without messing with environment variables. If this is the case, then it should just be picked up automatically by `System.loadLibrary("accumulo")`. Messing with `LD_LIBRARY_PATH` directly is just a hack to get `System.loadLibrary()` to find it from a standalone "tarball" packaging.

However, `System.loadLibrary()` isn't always recommended. In some systems, the file naming conventions won't match up to the name of the files. For example, on Macs, it can be a problem if the filename is `libaccumulo.jnilib` instead of `libaccumulo.dylib` or vice-versa. In those cases, it's not enough to specify the path. You have to specify the exact file. This is also the way it is supposed to be done for all JNI libs according to the Fedora/EPEL/RHEL packaging guidelines for Java. In such cases, you have to have some mechanism to specify the files directly, so you can use `System.load(filename)` instead of `System.loadLibrary(libname)`. Hence the property. We can inject specific filenames (but we also allow directories that we'll search) using the value of the Java system property `accumulo.native.lib.path`.